### ISSUE TITLE
java_opts not being picked up from wrapper recipe

### ISSUE THREAD

**Issue by dnataraj** [bdwyertech/chef-wildfly#22]
First of all, many thanks for your previous commit in order to provision domain.conf - which you went out of your way to do since it is clear that the cookbook is for "standalone".

I just can't seem to override java_opts though (the defaults you have provided in java_opts.rb). At the top of my wrapper recipe, I have : 
node.override['wildfly']['java_opts']['xmx'] = '1024m'
node.override['wildfly']['java_opts']['xms'] = '512m'
(I have tried node.set, node.default etc in desperation)

I suspect that I am doing something wrong here - do I need to have an attributes/java_opts.rb with overrides for this to work?

**Comment by bdwyertech** [comment]
I would give the attributes a shot. Although many say we should be providing attributes and such in-line, I would still override them from an attributes file.  Below, I show how I am overriding these in my own deployments.

```ruby
# Encoding: UTF-8

# => Default Wildfly Java Options
# =>  JAVA_OPTS="-Xms64m -Xmx512m -XX:MaxPermSize=256m -Djava.net.preferIPv4Stack=true"
# =>  JAVA_OPTS="$JAVA_OPTS -Djboss.modules.system.pkgs=$JBOSS_MODULES_SYSTEM_PKGS -Djava.awt.headless=true"


# => The defaults are adjustable here
default['wildfly']['java_opts']['xmx'] = ((node['memory']['total'].to_i / 1024) * 0.6).round.to_s + 'm'
default['wildfly']['java_opts']['xms'] = (((node['memory']['total'].to_i / 1024) * 0.6) / 8).round.to_s + 'm'
default['wildfly']['java_opts']['xx_maxpermsize'] = (((node['memory']['total'].to_i / 1024) * 0.6) / 2).round.to_s + 'm'
default['wildfly']['java_opts']['preferipv4'] = 'true'
default['wildfly']['java_opts']['headless'] = 'true'

# => Specify other java options in this space-deliniated array.
default['wildfly']['java_opts']['other'] = %w{
-Dorg.jboss.as.logging.per-deployment=false
}
```

**Comment by dnataraj** [comment]
I got this to work! Thanks. My next issues is figuring out why running opt/wildfly/bin/domain.sh starts up a server group, but starting wildfly service does not - but I'll try and solve that. Thanks for all your insights.

**Comment by bdwyertech** [comment]
I'd take a look at `/etc/default/wildfly.conf`, this is where many variables are set which the init script picks up, one of which is standalone vs domain mode @dnataraj

**Comment by dnataraj** [comment]
@bdwyertech I got my domain configured and running with your cookbook. Your wildfly.conf template is correct - I just had to introduce my own host.xml into play with node.set['wildfly']['dom']['host_conf'] = 'host.xml' - otherwise the recipe was (correctly) using host-master.xml which does *not* have the server groups defined - hence obviously, servers were not starting up! So I just did a 

    cookbook_file "/opt/wildfly/domain/configuration/host-master.xml" do
        source "host-master.xml"
        mode "0644"
    end

and everything is good. Thanks!
